### DISCUSSION TITLE
installing forge 1.12.2 via portainer

### DISCUSSION THREAD

**Discussion by DevDroid1** [itzg/docker-minecraft-server#2587] [Q&A] [answered]
hey i am trying to use docker to add a minecraft server container to my Synology NASand I specifically need version 1.12.2 with forge installed however I am unsure how to stop it from auto-updating to the latest version I really need this soon for an event and I can't find any answers online that show how to do it on portainer any suggestion please add web script if you have figured it out

for reference the current script I m using is the following
version: "3.8"
services:
  minecraft:
    image: itzg/minecraft-server
    container_name: minecraftjava
    environment:
      - UID=1024
      - GID=100
      - TYPE=FORGE
      - SERVER_NAME=HDROIDSERVER
      - JVM_XX_OPTS=-XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:InitiatingHeapOccupancyPercent=15
      - MAX_MEMORY=8G
      - EULA=TRUE
    volumes:
      - /volume1/docker/minecraftjava:/data:rw
    network_mode: host
    restart: always

**Comment by itzg** [comment]
Set and pin the Minecraft version (1.12.2) with VERSION env var

https://docker-minecraft-server.readthedocs.io/en/latest/versions/minecraft/

Can pin Forge version with FORGE_VERSION env var if needed

https://docker-minecraft-server.readthedocs.io/en/latest/types-and-platforms/server-types/forge/

**Comment by DevDroid1** [comment]
that resolved the problem that i had itzg

however now it can not get past the starting sequence on portainer
if I look at my logs it seems to keep repeating the same actions and I end up with a last output of 
failed to ping localhost:25565 : could not connect to Minecraft server: dial tcp [::1]:25565: connect: connection refused

here are my logs as I said they keep repeating
A problem occurred running the Server launcher.java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraftforge.fml.relauncher.ServerLaunchWrapper.run(ServerLaunchWrapper.java:70)
	at net.minecraftforge.fml.relauncher.ServerLaunchWrapper.main(ServerLaunchWrapper.java:34)
Caused by: java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader (jdk.internal.loader.ClassLoaders$AppClassLoader and java.net.URLClassLoader are in module java.base of loader 'bootstrap')
	at net.minecraft.launchwrapper.Launch.<init>(Launch.java:34)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	... 6 more
2024-01-12T17:57:58.310Z	WARN	mc-server-runner	Minecraft server failed. Inspect logs above for errors that indicate cause. DO NOT report this line as an error.	{"exitCode": 1}
2024-01-12T17:57:58.310Z	INFO	mc-server-runner	Done
[init] Running as uid=1024 gid=100 with /data as 'drwxr-xr-x 6 1024 100 4096 Jan 12 17:39 /data'
[init] Resolving type given FORGE
[mc-image-helper] 17:58:02.043 INFO  : Forge version 14.23.5.2859 for minecraft version 1.12.2 is already installed
[mc-image-helper] 17:58:05.119 INFO  : Created/updated 1 property in /data/server.properties

**Reply by itzg** [reply]
Use a Java 8 tagged image 

https://docker-minecraft-server.readthedocs.io/en/latest/versions/java/

Please read the documentation 😀

**Reply by DevDroid1** [reply]
i says that the java version is java 8 but I still get the same error should I use a specific version

**Reply by DevDroid1** [reply]
i need this server up urgently for a friend and the deadline is this evening (in the UK)

**Reply by itzg** [reply, marked as answer]
Oh...your comment changed 😀 and now I'm finding myself changing my comment. The logs are saying it is Java 8 but you're getting the exact same failure? Or is it something different? Please provide new logs. 

I'll leave this for the old comment:

It'll be the **image** field -- you'll probably need to recreate the container since that is the first field it's probably asking for. Set it to `itzg/minecraft-server:java8-multiarch`

https://docs.portainer.io/user/docker/containers/add#image-configuration-section

**Reply by itzg** [reply]
For comparison, the key phrase in your original logs

```
Caused by: java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader 
```

**Comment by DevDroid1** [comment]
Thank you very much that has helped me out so much thank you for the quick reply and the advice
